pyemma/coordinates/clustering/interface.py

```diff
@@ -23,30 +23,52 @@ def __init__(self, metric='euclidean'):
         super(AbstractClustering, self).__init__()
         self.metric = metric
         self.clustercenters = None
-        self.dtrajs = []
+        self._dtrajs = []
+
+    @property
+    def dtrajs(self):
+        if (len(self._dtrajs) == 0):  # nothing assigned yet, doing that now
+            return self.assign(stride = 1)
+        else:
+            return self._dtrajs  # returning what we have saved
 
     def _map_array(self, X):
         """get closest index of point in :attr:`clustercenters` to x."""
-        dtraj = np.empty(X.shape[0], np.int64)
+        dtraj = np.empty(X.shape[0], dtype=self.output_type())
```

Comment:
I'm not sure if the underlying c code does access arbitrary types from output_type. At least an assertion should go here, that currently only int64 is supported.

    if(PyArray_TYPE(np_dtraj)!=NPY_INT64) { PyErr_SetString(PyExc_ValueError, "dtype of \"dtraj\" isn\'t int (64)."); goto error; };